test/extensions/common/aws/credentials_provider_impl_test.cc

```diff
@@ -62,6 +91,114 @@ TEST_F(EvironmentCredentialsProviderTest, NoSessionToken) {
+    TestEnvironment::setEnvVar("AWS_PROFILE", profile, 1);
+  }
+
+  Event::SimulatedTimeSystem time_system_;
+  Api::ApiPtr api_;
+  CredentialsFileCredentialsProvider provider_;
+};
+
+TEST_F(CredentialsFileCredentialsProviderTest, FileDoesNotExist) {
+  TestEnvironment::setEnvVar("AWS_SHARED_CREDENTIALS_FILE", "/file/does/not/exist", 1);
+
+  const auto credentials = provider_.getCredentials();
+  EXPECT_FALSE(credentials.accessKeyId().has_value());
+  EXPECT_FALSE(credentials.secretAccessKey().has_value());
+  EXPECT_FALSE(credentials.sessionToken().has_value());
+}
+
+TEST_F(CredentialsFileCredentialsProviderTest, ProfileDoesNotExist) {
+  setUpTest(CREDENTIALS_FILE_CONTENTS, "invalid_profile");
+
+  const auto credentials = provider_.getCredentials();
+  EXPECT_FALSE(credentials.accessKeyId().has_value());
+  EXPECT_FALSE(credentials.secretAccessKey().has_value());
+  EXPECT_FALSE(credentials.sessionToken().has_value());
+}
+
+TEST_F(CredentialsFileCredentialsProviderTest, IncompleteProfile) {
+  setUpTest(CREDENTIALS_FILE_CONTENTS, "profile2");
+
+  const auto credentials = provider_.getCredentials();
+  EXPECT_EQ("profile2_access_key", credentials.accessKeyId().value());
+  EXPECT_FALSE(credentials.secretAccessKey().has_value());
+  EXPECT_FALSE(credentials.sessionToken().has_value());
+}
+
+TEST_F(CredentialsFileCredentialsProviderTest, DefaultProfile) {
+  setUpTest(CREDENTIALS_FILE_CONTENTS, "");
+
+  const auto credentials = provider_.getCredentials();
+  EXPECT_EQ("default_access_key", credentials.accessKeyId().value());
+  EXPECT_EQ("default_secret", credentials.secretAccessKey().value());
+  EXPECT_EQ("default_token", credentials.sessionToken().value());
+}
+
+TEST_F(CredentialsFileCredentialsProviderTest, CompleteProfile) {
+  setUpTest(CREDENTIALS_FILE_CONTENTS, "profile1");
+
+  const auto credentials = provider_.getCredentials();
+  EXPECT_EQ("profile1_acc=ess_key", credentials.accessKeyId().value());
+  EXPECT_EQ("profile1_secret", credentials.secretAccessKey().value());
+  EXPECT_EQ("profile1_token", credentials.sessionToken().value());
+}
+
+TEST_F(CredentialsFileCredentialsProviderTest, EmptySecret) {
+  setUpTest(CREDENTIALS_FILE_CONTENTS, "profile3");
+
+  const auto credentials = provider_.getCredentials();
+  EXPECT_EQ("profile3_access_key", credentials.accessKeyId().value());
+  EXPECT_FALSE(credentials.secretAccessKey().has_value());
+  EXPECT_FALSE(credentials.sessionToken().has_value());
+}
+
+TEST_F(CredentialsFileCredentialsProviderTest, SpacesBetweenParts) {
+  setUpTest(CREDENTIALS_FILE_CONTENTS, "profile4");
+
+  const auto credentials = provider_.getCredentials();
+  EXPECT_EQ("profile4_access_key", credentials.accessKeyId().value());
+  EXPECT_EQ("profile4_secret", credentials.secretAccessKey().value());
+  EXPECT_EQ("profile4_token", credentials.sessionToken().value());
+}
+
+TEST_F(CredentialsFileCredentialsProviderTest, EmptyCredsCached) {
```

Comment:
I guess the name of the test can change from `EmptyCredsCached` to something that indicate that the credentials expired?

Or other option is to add a separate unit test that will load a valid credentials first instead of an empty credentials. And then after credentials expired if the credentials file or profile is updated it will pick the new credentials.

Reply:
This has been addressed.